Revit_Core_Engine/Query/Bounds.cs

```diff
@@ -45,16 +46,36 @@ public static BoundingBoxXYZ Bounds(this IEnumerable<Solid> solids, Transform tr
             if (solids == null || !solids.Any())
                 return null;
 
-            Solid union = solids.First();
-            foreach (Solid solid in solids.Skip(1))
+            double minX = double.MaxValue;
+            double minY = double.MaxValue;
+            double minZ = double.MaxValue;
+            double maxX = double.MinValue;
+            double maxY = double.MinValue;
+            double maxZ = double.MinValue;
+
+            foreach (Solid solid in solids)
             {
-                union = BooleanOperationsUtils.ExecuteBooleanOperation(union, solid, BooleanOperationsType.Union);
+                BoundingBoxXYZ solidBBox = solid.GetBoundingBox();
+                XYZ solidMin = solidBBox.Min;
+                XYZ solidMax = solidBBox.Max;
+                XYZ solidOrigin = solidBBox.Transform.Origin;
+
+                minX = Math.Min(minX, solidMin.X + solidOrigin.X);
+                minY = Math.Min(minY, solidMin.Y + solidOrigin.Y);
+                minZ = Math.Min(minZ, solidMin.Z + solidOrigin.Z);
+                maxX = Math.Max(maxX, solidMax.X + solidOrigin.X);
+                maxY = Math.Max(maxY, solidMax.Y + solidOrigin.Y);
+                maxZ = Math.Max(maxZ, solidMax.Z + solidOrigin.Z);
             }
 
+            BoundingBoxXYZ unionBbox = new BoundingBoxXYZ();
+            unionBbox.Min = new XYZ(minX, minY, minZ);
+            unionBbox.Max = new XYZ(maxX, maxY, maxZ);
+
             if (transform != null)
-                union = SolidUtils.CreateTransformed(union, transform);
+                unionBbox.Transform = transform;
 
-            return union.GetBoundingBox();
+            return unionBbox;
```

Comment:
Yes that's right. I will change that to keep the previous way (transform the solid -> convert to bounding box) to avoid different coordinate systems and issues with bounding box transform.